File: U1/EXAMEN_SOAP_DOTNET_GR01/01 SERVIDOR/BanquitoServer_Soap_DotNet_GR01/BanquitoServer_Soap_DotNet_GR01/DataAccess/Repositories/CreditoRepository.cs
using System;
using System.Data.Entity.Validation;
using System.Linq;
using System.Text;
using BanquitoServer_Soap_DotNet_GR01.Models;

namespace BanquitoServer_Soap_DotNet_GR01.DataAccess.Repositories
{
    /// <summary>
    /// Repositorio para operaciones de Crédito
    /// </summary>
    public class CreditoRepository
    {
        private readonly BanquitoDbContext _context;

        public CreditoRepository()
        {
            _context = new BanquitoDbContext();
        }

        public CreditoRepository(BanquitoDbContext context)
        {
            _context = context;
        }

        /// <summary>
        /// Verificar si el cliente tiene un crédito activo
        /// </summary>
        public bool TieneCreditoActivo(string cedula)
        {
            var tieneCredito = (from cr in _context.Creditos
                               join cl in _context.Clientes on cr.ClienteId equals cl.ClienteId
                               where cl.Cedula == cedula
                                  && cr.Estado == "ACTIVO"
                               select cr).Any();

            return tieneCredito;
        }

        /// <summary>
        /// Guardar un crédito con su tabla de amortización
        /// </summary>
        public void Save(Credito credito)
        {
            try
            {
                _context.Creditos.Add(credito);
                _context.SaveChanges();
            }
            catch (DbEntityValidationException ex)
            {
                // Construir mensaje detallado de errores de validación
                StringBuilder sb = new StringBuilder();
                sb.AppendLine("Errores de validación de Entity Framework:");

                foreach (var validationErrors in ex.EntityValidationErrors)
                {
                    foreach (var validationError in validationErrors.ValidationErrors)
                    {
                        sb.AppendLine($"  - Propiedad: {validationError.PropertyName}, Error: {validationError.ErrorMessage}");
                    }
                }

                throw new Exception(sb.ToString(), ex);
            }
        }

        /// <summary>
        /// Buscar crédito por número de crédito
        /// </summary>
        public Credito FindByNumeroCredito(string numeroCredito)
        {
            return _context.Creditos
                .Include("CuotasAmortizacion")
                .FirstOrDefault(c => c.NumeroCredito == numeroCredito);
        }
    }
}

File: U2/PAO202551_ELECTRO_BDDNUBE_RESTFULL_JAVA_GR01/04. SERVIDOR/BanquitoServer_Restfull_Java_GR01/src/main/java/ec/edu/monster/Repositories/CreditoRepository.java
package ec.edu.monster.Repositories;

import ec.edu.monster.Models.Credito;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;
import java.util.Optional;

/**
 * Repositorio para operaciones de Crédito
 */
@Repository
public interface CreditoRepository extends JpaRepository<Credito, Long> {

    /**
     * Verificar si el cliente tiene un crédito activo
     */
    @Query("SELECT CASE WHEN COUNT(cr) > 0 THEN true ELSE false END FROM Credito cr " +
            "JOIN cr.cliente cl " +
            "WHERE cl.cedula = :cedula " +
            "AND cr.estado = 'ACTIVO'")
    boolean tieneCreditoActivo(@Param("cedula") String cedula);

    /**
     * Buscar crédito por número de crédito
     */
    @Query("SELECT cr FROM Credito cr " +
            "LEFT JOIN FETCH cr.cuotasAmortizacion " +
            "WHERE cr.numeroCredito = :numeroCredito")
    Optional<Credito> findByNumeroCredito(@Param("numeroCredito") String numeroCredito);
}
